### ISSUE TITLE
Add contributing guide

### ISSUE THREAD

**Issue by cirocosta** [concourse/concourse-bosh-release#67]
# Feature Request

## What challenge are you facing?

There are certain steps that we need to go through before submitting a PR to this repository, e.g., ensuring that the `erb` templates were properly generated.

Currently, that's "tribal knowledge" spread across the team that is not openly communicated (and sometimes, forgotten by us 😅 ) 

## A Modest Proposal

Have a `CONTRIBUTING.md` file, just like we do in `concourse/concourse` (see [`contributing.md`](https://github.com/concourse/concourse/blob/9b9c15e5af33ab9c4176875e5485b48397b91e12/CONTRIBUTING.md)).

**Comment by cirocosta** [comment]
Oh, I just realized that @taylorsilva  just had https://github.com/concourse/concourse-bosh-release/pull/65 in 🤔

**Comment by taylorsilva** [comment]
This is a great idea if there are other things to capture. Running that script for the job templates is only one thing, but we also have the entire `/dev` directory that could go into a `contributing.md`, maybe?

What other items could go into a `contributing.md`?

**Comment by cirocosta** [comment]
yeah, perhaps documenting the `dev` stuff would be good indeed, e.g., how we document how to build and run Concourse in the main contributinig one:

> After making any changes, you can try them out by rebuilding and recreating the web and worker containers:
> `$ docker-compose up --build -d`
> [...]

(see https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md#rebuilding-to-test-your-changes)

**Comment by cirocosta** [comment]
oh, something else:

- https://github.com/concourse/concourse-bosh-release/pull/66#pullrequestreview-312596990

giving some guidance on adding things like `example:`, descriptions, not setting defaults, etc would be pretty nice
